src/libraries/Common/src/System/Security/Cryptography/SLHDsa.cs

```diff
@@ -0,0 +1,1620 @@
+        public int ExportSLHDsaPublicKey(Span<byte> destination)
+        {
+            ThrowIfDisposed();
+
+            if (destination.Length < PublicKeySizeInBytes)
+            {
+                throw new ArgumentException(nameof(destination), SR.Argument_DestinationTooShort);
+            }
+
+            ExportSLHDsaPublicKeyCore(destination.Slice(0, PublicKeySizeInBytes));
+            return PublicKeySizeInBytes;
+        }
+
+        /// <summary>
+        ///  Exports the current key in the FIPS 205 secret key format.
+        /// </summary>
+        /// <param name="destination">
+        ///   The buffer to receive the secret key.
+        /// </param>
+        /// <returns>
+        ///   The number of bytes written to <paramref name="destination"/>.
+        /// </returns>
+        /// <exception cref="ArgumentException">
+        ///   <paramref name="destination"/> is too small to hold the secret key.
+        /// </exception>
+        /// <exception cref="CryptographicException">
+        ///   An error occurred while exporting the key.
+        /// </exception>
+        public int ExportSLHDsaSecretKey(Span<byte> destination)
+        {
+            ThrowIfDisposed();
+
+            if (destination.Length < SecretKeySizeInBytes)
+            {
+                throw new ArgumentException(nameof(destination), SR.Argument_DestinationTooShort);
+            }
+
+            ExportSLHDsaSecretKeyCore(destination.Slice(0, SecretKeySizeInBytes));
+            return SecretKeySizeInBytes;
+        }
+
+        /// <summary>
+        ///  Exports the private seed of the current key.
+        /// </summary>
+        /// <param name="destination">
+        ///   The buffer to receive the private seed.
+        /// </param>
+        /// <returns>
+        ///   The number of bytes written to <paramref name="destination"/>.
+        /// </returns>
+        /// <exception cref="ArgumentException">
+        ///   <paramref name="destination"/> is too small to hold the private seed.
+        /// </exception>
+        /// <exception cref="CryptographicException">
+        ///   An error occurred while exporting the private seed.
+        /// </exception>
+        public int ExportSLHDsaPrivateSeed(Span<byte> destination)
+        {
+            ThrowIfDisposed();
+
+            int privateSeedSizeInBytes = _parameterSetInfo.PrivateSeedSizeInBytes;
+            if (destination.Length < privateSeedSizeInBytes)
+            {
+                throw new ArgumentException(nameof(destination), SR.Argument_DestinationTooShort);
+            }
+
+            ExportSLHDsaPrivateSeedCore(destination.Slice(0, privateSeedSizeInBytes));
+            return privateSeedSizeInBytes;
+        }
+
+        /// <summary>
+        ///   Generates a new SLH-DSA-SHA2-128s key.
+        /// </summary>
+        /// <returns>
+        ///   The generated key.
+        /// </returns>
+        public static SLHDsa GenerateSLHDsaSha2_128sKey()
```

Comment:
SLH-DSA has too many options, this should definitely be parameterized.  (See the draft API proposal at https://github.com/dotnet/runtime/issues/113506)